ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py

```diff
@@ -1611,3 +1611,54 @@ def test_tucker_tensorly(tol_norm_2, tol_max_abs, shape, ranks):
         ivy.max(ivy.abs(rec_svd - rec_random)) < tol_max_abs,
         "abs norm of difference between svd and random init too high",
     )
+
+
+# dot
+@st.composite
+def _generate_dot_dtype_and_arrays(draw):
+    input_dtype = [draw(st.sampled_from(["float32", "float64"]))]
+    print()
+    matrices_dims = draw(
+        st.lists(st.integers(min_value=2, max_value=10), min_size=4, max_size=4)
+    )
+    shape_1 = (matrices_dims[0], matrices_dims[1])
+    shape_2 = (matrices_dims[1], matrices_dims[2])
+
+    matrix_1 = draw(
+        helpers.dtype_and_values(
+            shape=shape_1,
+            dtype=input_dtype,
+            min_value=-10,
+            max_value=10,
+        )
+    )
+    matrix_2 = draw(
+        helpers.dtype_and_values(
+            shape=shape_2,
+            dtype=input_dtype,
+            min_value=-10,
+            max_value=10,
+        )
```

Comment:
Why are we only generating matrices? If you look at the documentation of `np.dot` ( https://numpy.org/doc/stable/reference/generated/numpy.dot.html), you'll see that it supports the inputs to have any number of dimensions. Can we please update the test to generate tensors of arbitrary no. of dimensions as well and in case our implementations do not support this, can you update them as well?